### PR TITLE
Docs: document apostrophe as an explicit string literal trigger

### LINKED ISSUE #960: Docs: document apostrophe as an explicit string literal trigger
### Description

It's an undocumented feature

By starting the cell content with `'` character (apostrophe) our parser treats the rest of it as string. It is a way to prevent automatic conversion strings to number, date and time values.

### Links
https://handsoncode.slack.com/archives/C031GBX62SE/p1667382294314379 (internal discussion)
https://github.com/handsontable/hyperformula/discussions/959
https://github.com/handsontable/hyperformula/blob/fd0a40581d1dea2a19c2307228825ad0c3b6dedb/test/CellContentParser.spec.ts#L128-L135

### Type of issue

- [x] Missing documentation
- [ ] Error in the documentation
- [ ] Request for a demo
- [ ] Request for a tutorial, guide etc.
- [ ] Other issues, suggestions or ideas

### PR DESCRIPTION
**Pull request by sequba**
### Context

Added section 'Forcing the string value type' to the 'Types of values' guide.

Unit tests have been added in https://github.com/handsontable/hyperformula/pull/1161

### How did you test your changes?

- [x] unit tests pass
- [x] local docs build

### Types of changes

- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [ ] New feature or improvement (a non-breaking change that adds functionality)
- [ ] Bug fix (a non-breaking change that fixes an issue)
- [ ] Additional language file, or a change to an existing language file (translations)
- [x] Change to the documentation

### Related issues:
Fixes #960 

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to HyperFormula](https://hyperformula.handsontable.com/guide/contributing.html) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://goo.gl/forms/yuutGuN0RjsikVpM2).
- [x] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard.
- [x] My change is compatible with Microsoft Excel.
- [x] My change is compatible with Google Sheets.
- [ ] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file.
- [ ] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

File: docs/guide/types-of-values.md [hunk 1]
```diff
@@ -30,6 +30,28 @@ operations such as calculating the number of days between two dates.
 - A DateTime value is represented as the number of (possibly fractional) days
   since [`nullDate`](../api/interfaces/configparams.md#nulldate).
 
+## Forcing the string value type
+
+Similarly to other popular spreadsheet software, HyperFormula automatically detects the type of an input value.
+On some occasions, the value should be treated as a string even though it's parsable as a number/date/formula/etc.
```
Comment (evanSe) [inline docs/guide/types-of-values.md:36]:
```suggestion
On some occasions, the value should be treated as a string even though it's parsable as a number, date, formula, etc.
```

File: docs/guide/types-of-values.md [hunk 2]
```diff
@@ -30,6 +30,28 @@ operations such as calculating the number of days between two dates.
...
+The typical examples are numeric values with no number semantics, such as zip codes, bank sort codes, social security numbers, etc.
```
Comment (evanSe) [inline docs/guide/types-of-values.md:37]:
```suggestion
Typical examples are numeric values with no number semantics, such as zip codes, bank sort codes, social security numbers, etc.
```

File: docs/guide/types-of-values.md (hunk 1 above)
Comment (evanSe) [reply docs/guide/types-of-values.md:36]: Also wondering about the "etc." can we be more explicit?

Comment (sequba) [reply docs/guide/types-of-values.md]: Changed to the complete list of cell value types.

**Review by evanSe — APPROVED** [review]

**Pull request merged**